torch/distributed/_tensor/ops/collectives_ops.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+# implement matrix related ops for distributed tensor
+
+import torch
+from torch.distributed._tensor.op_schema import OpSchema, OutputSharding
+from torch.distributed._tensor.ops.utils import register_prop_rule
+from torch.distributed._tensor.placement_types import DTensorSpec
+
+_c10d_functional = torch.ops._c10d_functional
+
+
+@register_prop_rule(_c10d_functional.all_to_all_single.default)
```

Comment:
Can you elaborate why you need this?

DTensor works on top of collectives, so it calls collectives under the hood to transform a tensor sharding layout. Letting DTensor to work with the collective op directly is a pretty confusing behavior IMO. For example: why we keep the original sharding if DTensor runs with the alltoall_single op?

Reply:
The idea here is to allow mixing parallelism strategies -- i.e. supporting using DTensor for tensor parallelism while being able to use the collectives for higher level operations.

I was thinking about how to combine tensor parallelism with ring attention -- with this we end up rotating the kv blocks but still preserve the sharding information for tensor parallelism

I'm not sure this is the best direction to go by any means. I did this as an experiment after thinking about how you could compose DTensor and collectives after chatting with @wconstab last week